sensorimotor/experiments/capacity/experiment.py

```diff
@@ -44,179 +44,252 @@
+                                      showReset=True,
+                                      resetShading=PLOT_RESET_SHADING)
+      runner.tm.mmGetCellActivityPlot(title=title,
+                                      activityType="predictedActiveCells",
+                                      showReset=True,
+                                      resetShading=PLOT_RESET_SHADING)
+      runner.tp.mmGetCellActivityPlot(title=title, showReset=True,
+                                      resetShading=PLOT_RESET_SHADING)
 
-    if PLOT >= 1:
-      title = "worlds: {0}, elements: {1}".format(numWorlds, numElements)
-      runner.tp.mmGetPlotConnectionsPerColumn(title=title)
-      runner.tp.mmGetCellActivityPlot(title=title)
-
-    print "Testing (worlds: {0}, elements: {1})...".format(numWorlds,
-                                                           numElements)
-    sequences = runner.generateSequences(completeSequenceLength / 4,
-                                         randomAgents,
-                                         verbosity=VERBOSITY,
-                                         numSequences=4)
-    runner.feedLayers(sequences, tmLearn=False, tpLearn=False,
-                      verbosity=VERBOSITY,
-                      showProgressInterval=SHOW_PROGRESS_INTERVAL)
-    print "Done testing.\n"
-
-    if VERBOSITY >= 2:
-      print "Overlap:"
-      print
-      print runner.tp.mmPrettyPrintDataOverlap()
-      print
-
-    print MonitorMixinBase.mmPrettyPrintMetrics(
-      runner.tp.mmGetDefaultMetrics() + runner.tm.mmGetDefaultMetrics())
-    print
 
-    elapsed = int(time.time() - start)
-    print "Total time: {0:2} seconds.".format(elapsed)
 
+def writeOutput(outputDir, runner, numElems, numWorlds, elapsedTime):
+  if not os.path.exists(outputDir):
+    os.makedirs(outputDir)
+  fileName = "{0:0>3}x{1:0>3}.csv".format(numWorlds, numElems)
+  filePath = os.path.join(outputDir, fileName)
+  with open(filePath, "wb") as outputFile:
+    csvWriter = csv.writer(outputFile)
     header = ["# worlds", "# elements", "duration"]
-    row = [numWorlds, numElements, elapsed]
-
+    row = [numWorlds, numElems, elapsedTime]
     for metric in (runner.tp.mmGetDefaultMetrics() +
                    runner.tm.mmGetDefaultMetrics()):
       header += ["{0} ({1})".format(metric.prettyPrintTitle(), x) for x in
-                 ["min", "max", "sum", "mean", "stddev"]]
-      row += [metric.min, metric.max, metric.sum,
-              metric.mean,  metric.standardDeviation]
-
+                ["min", "max", "sum", "mean", "stddev"]]
+      row += [metric.min, metric.max, metric.sum, metric.mean,
+              metric.standardDeviation]
     csvWriter.writerow(header)
     csvWriter.writerow(row)
-    csvFile.flush()
+    outputFile.flush()
+
 
-  if PLOT >= 1:
+
+def run(numWorlds, numElems, paramsPath, outputDir, plotVerbosity,
+        consoleVerbosity, params=None):
+  if params is None:
```

Comment:
Do this in `__main__` instead, remove `paramsPath`, and require `params` to be passed in to this function.